### PR TITLE
Add Fake.Net.Http documentation & API update

### PR DESCRIPTION
**Pull request by vlukash**
* Add Fake.Net.Http documentation
* Change API to throw exceptions in case of error, so it's consistent with other APIs

### CODE REVIEW

**Comment by vlukash** [comment]
Related to #1746

**Comment by vlukash** [comment]
Seems like documentation is not autogenerated (from *.dll) for the netcore projects if *.fs files aren't included in the legacy FakeLib.fsproj. 

In this case, I don't see Fake.Net.Http namespace in the 
https://fake.build/apidocs/index.html

I can probably modify "GenerateDocs" target in build.fsx, or include *.fs files in the FakeLib.fsproj

@matthid, do you know what is the right way to autogenerate docs for netcore projects?

**Comment by matthid** [comment]
Actually, I think the easiest way is to just include it into fakelib until we have worked that out...

**Comment by vlukash** [comment]
Got it, thanks!
Updated FakeLib, ready for review

File: src/app/Fake.Net.Http/List.fs [hunk 1]
```diff
@@ -3,11 +3,13 @@
 open Fake.Net.Async
 open Fake.Net.Result
 
+/// [omit]
 // List extensions for traversing Result and Async types
 // Functions from fsharpforfunandprofit.com, please see details here:
 // https://fsharpforfunandprofit.com/posts/elevated-world-5/
 module List =
```
Comment (matthid) [inline src/app/Fake.Net.Http/List.fs]: Maybe we should mark that internal?

File: src/app/Fake.Net.Http/Result.fs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 namespace Fake.Net.Result
 
+/// [omit]
 module Result =
```
Comment (matthid) [inline src/app/Fake.Net.Http/Result.fs]: internal?

**Comment by matthid** [comment]
I think there is a place in build.fsx which needs to be edited if a new namespace (like `Fake.Net` here) is added. I'll try to put everything together

**Pull request merged**